server/internal/urn/resources.go

```diff
@@ -0,0 +1,307 @@
+package urn
+
+import (
+	"database/sql/driver"
+	"encoding/json"
+	"fmt"
+	"net/url"
+	"regexp"
+	"strings"
+
+	"github.com/speakeasy-api/gram/server/internal/constants"
+)
+
+type Resource struct {
+	Kind         ResourceKind
+	Source       string
+	SlugifiedURI string
+
+	checked bool
+	err     error
+}
+
+var (
+	nonSlugCharsRE    = regexp.MustCompile(`[^a-z0-9_-]+`)
+	multiDashRE       = regexp.MustCompile(`-+`)
+	multiUnderscoreRE = regexp.MustCompile(`_+`)
+)
+
+func sanitizeToSlug(s string) string {
```

Comment:
Is this different to the other sanitization / slugification logic we already have elsewhere? I'd rather we shared that